tests/test_metrics.py

```diff
@@ -0,0 +1,187 @@
+import unittest
+
+import numpy as np
+import pandas as pd
+
+from dna import utils
+from dna import metrics  
+
+
+class MetricsTestCase(unittest.TestCase):
+
+    def format_top_k(self, scores, ids, rank):
+        """
+        A helper function for the top k tests.
+        """
+        pipeline_dicts = [{"id": id} for id in ids]
+        actual_data = pd.DataFrame({"test_f1_macro": scores, "pipeline": pipeline_dicts})
+        ranked_data = pd.DataFrame({"rank": rank, "pipeline_id": ids})
+        return ranked_data, actual_data
+
+    def format_and_get_top_k(self, scores, ids, rank, k, top_k_function):
+        """
+        :param scores: the `scores` of the pipelines in a list
+        :param ids: a list of the pipeline ids
+        :param rank: a list of the pipeline rankings
+        :param k: the number of pipelines to get
+        :param top_k_function: the top k function to evaluate, top_k_regret or top_k_correct
+        """
+        ranked_data, actual_data = self.format_top_k(scores, ids, rank)
+        metric = top_k_function(ranked_data, actual_data, k)
+        return metric
+
+    def format_and_get_spearman(self, actual_data, ranked_data):
+        """
+        :param actual_data: a list of the real rankings
+        :param ranked_data: a list of the ranked predictions
+        """
+        actual_data.reverse() # our rank function in spearman will reverse this back
```

Comment:
If our spearman function is off by a sign, this seems like it hides the problem rather than fixing it. Is that what this does?

Reply:
Yes, we do need to fix the spearman function.  But how we fix it will decide what we need to do here.  See my comment in the PR opening
> Let's talk about it and fix it in a separate PR.

Reply:
It's better for the tests to be correct and fail, then to hide a bug in our code. That is the point. We can still fix it in a separate PR if you wish.